### DISCUSSION TITLE
File Upload Issue In Primefaces 12 Jakarta Version

### DISCUSSION THREAD

**Discussion by Santhanam12345** [primefaces/community#1742] [PrimeFaces] [answered]
Hello Primefaces Team ,

I am writing to inform you about an issue we have encountered with the file upload functionality in PrimeFaces 12 Jakarta version when running on Tomcat 10. The problem stems from the dependency on Apache Commons FileUpload, which utilizes javax.servlet.

As Jakarta EE compatibility is crucial for our application running on Tomcat 10, we are seeking assistance in resolving this file upload issue promptly. We understand that adjustments may be necessary to ensure seamless compatibility with the latest technologies.

Could you please provide us with a quote for addressing this issue? We are open to discussing the scope of changes required and any associated costs.

Thank you for your attention to this matter. We look forward to your prompt response and collaboration in resolving the file upload issue.

**Comment by Rapster** [comment, marked as answer]
Apache Fileupload is not jakartified AFAIR, and deprecated in PF, see https://primefaces.github.io/primefaces/14_0_0/#/../migrationguide/14_0_0?id=deprecated (unless you're using specifics from apache, no reason to stay on commons fileupload)

**Reply by Rapster** [reply]
Some details are also shared here: https://github.com/primefaces/primefaces/issues/10177

**Reply by Santhanam12345** [reply]
Hi Rapster thank you very much for your reply.. i just removed the following code from web.xml

<context-param> 
        <param-name>primefaces.UPLOADER</param-name> 
        <param-value>commons</param-value> 
    </context-param>

Now it is working fine..
